tests/workgroup/run_kernel.c

```diff
@@ -110,30 +110,55 @@ main (int argc, char **argv)
 
   size_t grid_size
       = global_work_size[0] * global_work_size[1] * global_work_size[2];
-  if (num_args == 1)
+  if (num_args > 0)
     {
```

Comment:
Does this branch accept arbitrary amount of arguments or just up to two? I see there is an assert in the else branch for checking there is up to two arguments but no check for 2+ arguments in this branch.